trillian_log_api.proto

```diff
@@ -182,23 +183,42 @@ service TrillianLog {
 
     // No direct equivalent at the storage level
     rpc GetInclusionProof (GetInclusionProofRequest) returns (GetInclusionProofResponse) {
+      option (google.api.http) = {
+        get: "/v1beta1/log/{log_id}/size/{tree_size}/inclusion/index/{leaf_index}"
```

Comment:
Regarding the URL scheme:

* Use plural for collections, ie, **/logs** instead of **/log** (https://cloud.google.com/apis/design/resource_names#collection_id)

* Size doesn't feel like a collection. There isn't a "Size" resource. I think **/logs/\_/leaves/\_** fits better, with size being a regular parameter when appropriate.

I would go for the following collections, using custom methods for most of the non-standard reads:

**/v1beta1/logs/{log_id}**
**/v1beta1/logs/{log_id}/leaves/{leaf_index}**
**/v1beta1/logs/{log_id}/roots**

That gives us the following:

GetInclusionProof:
GET **/v1beta1/logs/{log_id}/leaves/{leaf_index}:inclusion_proof**

GetInclusionProofByHash:
GET **/v1beta1/logs/{log_id}/leaves/{leaf_index}:inclusion_proof_by_hash**

GetConsistencyProof:
GET **/v1beta1/logs/{log_id}:consistency_proof**

GetLatestSignedLogRoot:
GET **/v1beta1/logs/{log_id}/roots:latest**

GetSequencedLeafCount:
GET **/v1beta1/logs/{log_id}/leaves:sequenced_count**

GetEntryAndProofRequest:
GET **/v1beta1/logs/{log_id}/leaves/{leaf_index}**

(I'm ignoring the methods you chose not to map as well.)

TrillianLog and TrillianMap's design is not resource oriented, so not all APIs map as nicely as I'd hope for. Ideally I'd like to give this more thought, but I guess we can go forward with the mapping as long as there's an understanding that it's not stable and is likely to change in the future. To be truly compliant to Google's API style we'd have to change most of our request and response messages as well.

Note that the paragraph above doesn't apply to the Admin (and soon to be) Quota APIs, which are resource oriented from the start and fit nicely in this model.